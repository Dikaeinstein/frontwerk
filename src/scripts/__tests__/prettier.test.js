/* eslint-disable global-require */
jest.mock('cross-spawn');
jest.mock('jest');

const cases = require('jest-in-case');

const testCases = [
  {
    name: 'calls prettier CLI with default args'
  },
  {
    name: '--no-write prevents --write argument from being added',
    args: ['--no-write']
  },
  {
    name: 'calls prettier CLI with args',
    args: ['my-src/**/*.js']
  },
  {
    name: 'does not use built-in config with .prettierrc file',
    fileExists: filename => filename === '.prettierrc'
  },
  {
    name: 'does not use built-in config with prettier.config.js file',
    fileExists: filename => filename === 'prettier.config.js'
  },
  {
    name: 'does not use built-in config with prettierrc pkg prop',
    hasPkgProp: prop => prop === 'prettierrc'
  },
  {
    name: 'does not use built-in ignore with --ignore-path',
    args: ['--ignore-path', './my-ignore']
  },
  {
    name: 'does not use built-in config with --config',
    args: ['--config', './custom-config.js']
  }
];

const testFn = ({
  fileExists = () => false,
  hasPkgProp = () => false,
  args = []
}) => {
  const { sync: crossSpawnSyncMock } = require('cross-spawn');
  const originalExit = process.exit;
  const originalArgv = process.argv;
  process.exit = jest.fn();

  Object.assign(require('../../utils/fileExists'), { fileExists });
  Object.assign(require('../../utils/pkg'), { hasPkgProp });
  Object.assign(require('../../utils/resolveBin'), {
    resolveBin: (modName, { executable = modName } = {}) => executable
  });

  process.exit = jest.fn();

  try {
    // tests
    process.argv = ['node', '../prettier', ...args];
    crossSpawnSyncMock.mockClear();

    require('../prettier');

    expect(crossSpawnSyncMock).toHaveBeenCalledTimes(1);
    const [firstCall] = crossSpawnSyncMock.mock.calls;
    const [script, calledArgs] = firstCall;
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot();
  } catch (error) {
    throw error;
  } finally {
    // afterEach
    process.exit = originalExit;
    process.argv = originalArgv;
    jest.resetModules();
  }
};

cases('format', testFn, testCases);
