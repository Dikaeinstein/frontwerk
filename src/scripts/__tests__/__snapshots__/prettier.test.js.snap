// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format --no-write prevents --write argument from being added 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js --no-write"`;

exports[`format calls prettier CLI with args 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js my-src/**/*.js"`;

exports[`format calls prettier CLI with default args 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js"`;

exports[`format does not use built-in config with .prettierrc file 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js"`;

exports[`format does not use built-in config with --config 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js --config ./custom-config.js"`;

exports[`format does not use built-in config with prettier.config.js file 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js"`;

exports[`format does not use built-in config with prettierrc pkg prop 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js"`;

exports[`format does not use built-in ignore with --ignore-path 1`] = `"node /Users/tricinel/Work/Personal/repos/ff-scripts/packages/frontwerk/src/scripts/format.js --ignore-path ./my-ignore"`;
